### PR TITLE
Add unit and integration tests for http lib

### PR DESCRIPTION
**Pull request by Thanhphan1147**
### Checklist

- [x] The [charm style guide](https://juju.is/docs/sdk/styleguide) was applied
- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The changes are compliant with [ISD054 - Managing Charm Complexity](https://discourse.charmhub.io/t/specification-isd014-managing-charm-complexity/11619)
- [x] The documentation is generated using `src-docs`
- [x] The documentation for charmhub is updated
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)

### CODE REVIEW

File: .github/workflows/integration_test.yaml [hunk 1]
```diff
@@ -8,15 +8,8 @@ jobs:
...
-      zap-target-port: 80
-      zap-rules-file-name: "zap_rules.tsv"
-      trivy-fs-enabled: true
-      trivy-image-config: "trivy.yaml"
+      provider: lxd
+      juju-channel: 3.3/stable
       self-hosted-runner: true
-      self-hosted-runner-label: "edge"
+      charmcraft-channel: latest/edge
+      modules: '["test_action.py", "test_charm.py"]'
```
Comment (yanksyoon) [inline .github/workflows/integration_test.yaml:15]: I think there's a missing newline here :)

File: .woke.yaml [hunk 2]
```diff
@@ -0,0 +1,6 @@
+ignore_files:
+  - lib/*
+
+rules:
+  # Ignore whitelist - we are using it to ignore pydantic in pyptoject.toml
+  - name: whitelist
```
Comment (yanksyoon) [inline .woke.yaml:6]: I think there's a missing newline here :)

File: README.md [hunk 3]
```diff
@@ -1,26 +1,31 @@
-<!--
-Avoid using this README file for information that is maintained or published elsewhere, e.g.:
+# Overview
 
-* metadata.yaml > published on Charmhub
-* documentation > published on (or linked to from) Charmhub
-* detailed contribution guide > documentation or CONTRIBUTING.md
+A Juju charm that deploys and manages haproxy on machine. HAProxy is a TCP/HTTP reverse proxy which is particularly suited for high availability environments. It features connection persistence through HTTP cookies, load balancing, header addition, modification, deletion both ways. It has request blocking capabilities and provides interface to display server status.
```
Comment (yanksyoon) [inline README.md:3]: s/manages haproxy on machine/manges haproxy on a VM/

File: requirements.txt [hunk 4]
```diff
@@ -1 +1,5 @@
-ops >= 2.2.0
+cryptography==43.0.0
+jinja2==3.1.4
+jsonschema==4.23.0
+ops==2.15.0
+pydantic==2.8.2
```
Comment (yanksyoon) [inline requirements.txt:5]: I think there's a missing new line here :D

**Review by yanksyoon — COMMENTED** [review]
Reviewd up to state/tls.py!

**Comment by Thanhphan1147** [comment]
closing as dups with #14

**Pull request closed without merging**